src/rp/index.ts

```diff
@@ -180,18 +183,25 @@ export class RelyingPartySolution {
 
     // TODO: [SIW-359] check all requeste claims of the requestedObj are satisfied
 
-    const vp_token = new SignJWT({ vp })
+    const vp_token = new SignJWT({
+      vp: vp,
+      jti: `${uuid.v4()}`,
+      iss: walletInstanceIdentifier,
+      nonce: requestObj.payload.nonce,
+    })
       .setAudience(requestObj.payload.response_uri)
+      .setIssuedAt()
       .setExpirationTime("1h")
       .setProtectedHeader({
         typ: "JWT",
         alg: "ES256",
```

Comment:
I don't want to be picky, given that the same problem also exists for obtaining the PID. But this value cannot be static but depends on the key that will be used for signing. We need an inverse function of [getKtyFromAlg](https://github.com/pagopa/io-react-native-jwt/blob/0cc8243be6f6093ebf11f7f36cc6375ce3cc997e/src/algorithms.ts#L18)

Reply:
Is this true for every signed jwt?

Reply:
yep!

Reply:
This is an issue occurring on multiple parts of the codebase like https://github.com/pagopa/io-react-native-wallet/commit/3683bcab054455459f239a6c59ec49a25532310e or https://github.com/pagopa/io-react-native-wallet/commit/0717493907935f6f9230b029642d160f15f6298d.

I made up my mind we need to handle the signing process in the `SignJWT` class, so that we don't have to pass intermediate values around the code. But it's just an idea.

I think we must handle the issue, but it's better we make a dedicated PR